spec/lib/subscriber_list/hash_digest_spec.rb

```diff
@@ -0,0 +1,53 @@
+RSpec.describe SubscriberList::HashDigest do
+  describe ".generate" do
+    subject { described_class.generate(input) }
+    let(:input) { {} }
+
+    it "returns nil when given an empty hash" do
+      expect(subject).to be_nil
+    end
+
+    context "when a hash is provided" do
+      let(:input) do
+        {
+          a: 1, b: 2, aircraft_type: %w(plane helicopter),
+          c: 3, d: 4, boat_type: %w(boat ship)
+        }
+      end
+
+      it "returns a digest for a hash" do
+        expect(subject).to eq("ee38bab7cf3dcfa3c0181cb26ad2bc25f006ca62fa030a81dc2b0ac353b4bddb")
+      end
+
+      it "returns the same digest every time the input is provided" do
+        digests = (1..5).map { described_class.generate(input) }
+        expect(digests.uniq.count).to eq 1
+      end
+    end
+
+    it "returns the same digest regardless of hash structure" do
+      first_input = {
+        a: [1, 2, 3, 4], b: %w(a b c d),
+        prior: 'one',
+        nested_tags: {
+          number: 2,
+          deep_nest: { foo: 'bar' },
+          more_tags: %w(more and more),
+        }
+      }
+      second_input = {
+        b: %w(a c b d), a: [1, 4, 3, 2],
+        nested_tags: {
+          more_tags: %w(more more and),
+          number: 2,
+          deep_nest: { foo: 'bar' },
+        },
+        prior: 'one'
+      }
+      expected_digest = "1528bd4ce9c8d03c240c87ddb7d44f5892e23136ffe692cd150ef4a75184aa77"
+
+      expect(described_class.generate(first_input)).to eq(expected_digest)
+      expect(described_class.generate(second_input)).to eq(expected_digest)
```

Comment:
I think it'd be reasonable here to just check these return matching digests rather than they both explicitly match a hardcoded digest.